modules/internal/ChapelBase.chpl

```diff
@@ -675,9 +675,11 @@ module ChapelBase {
 
   // Removing the 'eltType' arg results in errors for --baseline
   inline proc _ddata_shift(type eltType, data: _ddata(eltType), shift: integral) {
-    var ret: _ddata(eltType);
-     __primitive("shift_base_pointer", ret, data, shift);
-    return ret;
+    local {
```

Comment:
Why is this local block needed? If there's a good reason, can you add a comment above it to note why it's currently needed?